pkg/cmd/grafana-cli/commands/datamigrations/to_unified_storage.go

```diff
@@ -0,0 +1,210 @@
+package datamigrations
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	dashboard "github.com/grafana/grafana/pkg/apis/dashboard"
+	folders "github.com/grafana/grafana/pkg/apis/folder/v0alpha1"
+
+	authlib "github.com/grafana/authlib/types"
+	"github.com/grafana/grafana/pkg/apimachinery/identity"
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/utils"
+	"github.com/grafana/grafana/pkg/infra/db"
+	"github.com/grafana/grafana/pkg/infra/tracing"
+	"github.com/grafana/grafana/pkg/registry/apis/dashboard/legacy"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/storage/legacysql"
+	"github.com/grafana/grafana/pkg/storage/unified"
+	"github.com/grafana/grafana/pkg/storage/unified/parquet"
+	"github.com/grafana/grafana/pkg/storage/unified/resource"
+)
+
+// ToUnifiedStorage converts dashboards+folders into unified storage
+func ToUnifiedStorage(c utils.CommandLine, cfg *setting.Cfg, sqlStore db.DB) error {
+	ctx := authlib.WithAuthInfo(context.Background(), &identity.StaticRequester{
```

Comment:
@chaudyg / @stephaniehingtgen -- do we have a good approach to specify the config user via command line?  EG a token to use or something?

Reply:
You should use identity.WithServiceIdentitiy 
[example](https://github.com/grafana/grafana/blob/34b2cb5e02f66ea0a0d32903db483255fb93e730/pkg/apiserver/rest/dualwriter_syncer.go#L162-L163)

By default the unistore client is configure with the underlying instance token